source/s3_auto_ranged_get.c

```diff
@@ -381,28 +442,44 @@ static void s_s3_auto_ranged_get_request_finished(
 
     aws_s3_meta_request_lock_synced_data(meta_request);
 
-    ++auto_ranged_get->synced_data.num_parts_completed;
+    if (request->request_tag == AWS_S3_AUTO_RANGE_GET_REQUEST_TYPE_PART) {
+        ++auto_ranged_get->synced_data.num_parts_completed;
 
-    if (error_code == AWS_ERROR_SUCCESS) {
-        ++auto_ranged_get->synced_data.num_parts_successful;
+        if (error_code == AWS_ERROR_SUCCESS) {
+            ++auto_ranged_get->synced_data.num_parts_successful;
 
-        if (request->part_number == 1) {
-            AWS_ASSERT(num_parts > 0);
-            auto_ranged_get->synced_data.total_num_parts = num_parts;
-        }
+            if (request->part_number == 1) {
+                AWS_ASSERT(num_parts > 0);
+                auto_ranged_get->synced_data.total_num_parts = num_parts;
+            }
 
-        aws_s3_meta_request_stream_response_body_synced(meta_request, request);
+            aws_s3_meta_request_stream_response_body_synced(meta_request, request);
 
-        AWS_LOGF_DEBUG(
-            AWS_LS_S3_META_REQUEST,
-            "id=%p: %d out of %d parts have completed.",
-            (void *)meta_request,
-            (auto_ranged_get->synced_data.num_parts_successful + auto_ranged_get->synced_data.num_parts_failed),
-            auto_ranged_get->synced_data.total_num_parts);
-    } else {
-        ++auto_ranged_get->synced_data.num_parts_failed;
-
-        aws_s3_meta_request_set_fail_synced(meta_request, request, error_code);
+            AWS_LOGF_DEBUG(
+                AWS_LS_S3_META_REQUEST,
+                "id=%p: %d out of %d parts have completed.",
+                (void *)meta_request,
+                (auto_ranged_get->synced_data.num_parts_successful + auto_ranged_get->synced_data.num_parts_failed),
+                auto_ranged_get->synced_data.total_num_parts);
+        } else {
+            if (s_check_empty_file_download_error(request)) {
+                AWS_LOGF_DEBUG(
+                    AWS_LS_S3_META_REQUEST,
+                    "id=%p Getting an empty file, create a new request without range header to fetch the empty "
+                    "file",
+                    (void *)meta_request);
+                auto_ranged_get->synced_data.get_without_range = true;
+            } else {
+                ++auto_ranged_get->synced_data.num_parts_failed;
```

Comment:
I missed this--I think we should move this right after line 475 so that it increments for empty file download too (since the part did fail in this case).